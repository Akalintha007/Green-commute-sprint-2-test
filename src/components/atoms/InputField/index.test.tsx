import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import InputComponent from '.';

const handleClick = () => {
  console.info('You clicked the Chip.');
};

const handleDelete = () => {
  console.info('You clicked the delete icon.');
};

function renderInputField () {
  render(
    <InputComponent
      placeholder={'Search by job location'}
      handleClick={handleClick}
      handleDelete={handleDelete}
      values={["Hyderabad"]}
      eventHandler={function (location: string[]): void {
        throw new Error('Function not implemented.');
      }}
    />
  );
}

describe('Atoms', () => {
  it('InputField snapshot', () => {
    const tree = renderer
    .create(<InputComponent
      placeholder={'Search by job location'}
      handleClick={handleClick}
      handleDelete={handleDelete}
      eventHandler={function (location: string[]): void {
        throw new Error('Function not implemented.');
      }}
    />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});
  test('renders input field', () => {
    renderInputField();
    expect(screen.getByRole('combobox')).toBeInTheDocument();
  });

  test('renders chips', () => {
    renderInputField();
    const chipItem = screen.getByTestId('chip-item');
    expect(chipItem).toBeInTheDocument();
  });

  test('renders delete button on chip', () => {
    renderInputField();
    const deleteButton = screen.getByRole('button');
    expect(deleteButton).toBeInTheDocument();
  })
});
