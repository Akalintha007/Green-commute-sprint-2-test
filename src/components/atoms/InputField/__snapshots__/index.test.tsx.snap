// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms InputField snapshot 1`] = `
<div
  aria-owns={null}
  className="MuiAutocomplete-root css-16awh2u-MuiAutocomplete-root"
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-jfip22-MuiFormControl-root-MuiTextField-root"
  >
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiAutocomplete-inputRoot css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-invalid={false}
        autoCapitalize="none"
        autoComplete="off"
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input MuiAutocomplete-input MuiAutocomplete-inputFocused css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="tags-filled"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        placeholder="Search by job location"
        readOnly={false}
        required={false}
        role="combobox"
        spellCheck="false"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-hdw1oc"
        >
          <span
            className="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
